File: homemanager/sensors/sensor.py
from abc import ABC, abstractmethod
from enum import Enum


class SensorType(Enum):
    WEATHER = 1
    TEMPERATURE = 2
    HUMIDITY = 3

    def __repr__(self):
        str_repr = {
            SensorType.WEATHER: "WEATHER",
            SensorType.TEMPERATURE: "TEMPERATURE",
            SensorType.HUMIDITY: "HUMIDITY"
        }

        return str_repr[self]


class Sensor(ABC):
    @property
    def types(self):
        return set()

    @property
    def name(self):
        return ""

    @abstractmethod
    async def get_readings(self):
        ...

File: Common/Models/Sensor.cs
using System;
using System.Collections;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Data.Common;

namespace Common.Models
{
    public enum SensorType
    {
        Temperature,
        Humidity
    }

    public class Sensor : EntityBase
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int SensorId { get; set; }
        public String Name { get; set; }
        public String Location { get; set; }
        public SensorType Type { get; set; }
    }
}
